File: gdx2d-demoDesktop/src/ch/hevs/gdx2d/demos/selector/AboutDialog.java
package hevs.gdx2d.lib.gui;

import hevs.gdx2d.lib.Version;

import javax.swing.*;
import java.awt.*;
import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;
import java.net.URI;

/**
 * About dialog for the demo selector.
 *
 * @author Christopher Métrailler (mei)
 * @version 1.0
 */
public class AboutDialog {
	private final Icon icon = new ImageIcon(getClass().getResource("/lib/icon64.png"));
	private Component parent;
	private JPanel aboutPanel = new JPanel();

	private final static String TITLE = "About this application";

	private final static String INFO = "<HTML><BODY>"
			+ "<b>Demos selector for the \"gdx2d\" library.</b><br><br>"
			+ "Pierre-Andr&eacute; Mudry<br>"
			+ "Christopher M&eacute;trailler<br><br></BODY></HTML>";

	/**
	 * Create the about dialog with clickable links.
	 *
	 * @param parent the parent component
	 */
	public AboutDialog(Component parent) {
		this.parent = parent;

		aboutPanel.setLayout(new BoxLayout(aboutPanel, BoxLayout.Y_AXIS));
		aboutPanel.add(new JLabel(INFO));

		JLabel webLabel = new JLabel("Made for the ");
		addUrl(webLabel, "http://inf1.begincoding.net", "inf1 course");
		aboutPanel.add(webLabel);

		JLabel gitLabel = new JLabel();
		addUrl(gitLabel, "http://hevs-isi.github.io/gdx2d/", "http://hevs-isi.github.io/gdx2d/");
		aboutPanel.add(gitLabel);

		aboutPanel.add(new JLabel("<html><br><pre>" + Version.printVerbose() + "</pre></html>"));
	}

	/**
	 * Show the about dialog as a {@link JOptionPane}.
	 */
	public void setVisible() {
		JOptionPane.showMessageDialog(parent, aboutPanel, TITLE, JOptionPane.INFORMATION_MESSAGE, icon);
	}

	private static void addUrl(JLabel website, final String url, String text) {
		website.setText(String.format("<html>%s<a href=\"%s\">%s</a></html>", website.getText(), url, text));
		website.setCursor(new Cursor(Cursor.HAND_CURSOR));
		website.addMouseListener(new MouseAdapter() {
			@Override
			public void mouseClicked(MouseEvent e) {
				try {
					Desktop.getDesktop().browse(new URI(url));
				} catch (Exception ex) {
					System.err.println("Cannot open " + url);
					ex.printStackTrace();
				}
			}
		});
	}
}

File: gdx2d-demoDesktop/src_to_do/main/kotlin/ch/hevs/gdx2d/demos/selector/AboutDialog.kt
package ch.hevs.gdx2d.demos.selector

import ch.hevs.gdx2d.lib.Version

import javax.swing.*
import java.awt.*
import java.awt.event.MouseAdapter
import java.awt.event.MouseEvent
import java.net.URI

/**
 * About dialog for the demo selector.
 *
 * @author Christopher Métrailler (mei)
 * @version 1.0
 */
internal class AboutDialog
/**
 * Create the about dialog with clickable links.
 *
 * @param parent the parent component
 */
(private val parent: Component) {
    private val icon = ImageIcon(javaClass.getResource("/selector/icon64.png"))
    private val aboutPanel = JPanel()

    init {

        aboutPanel.preferredSize = Dimension(320, 200)
        aboutPanel.layout = BoxLayout(aboutPanel, BoxLayout.Y_AXIS)
        aboutPanel.add(JLabel(INFO))

        val webLabel = JLabel("Made for the ")
        addUrl(webLabel, "http://inf1.begincoding.net", "inf1 course")
        aboutPanel.add(webLabel)

        val gitLabel = JLabel()
        addUrl(gitLabel, "https://hevs-isi.github.io/gdx2d/", "https://hevs-isi.github.io/gdx2d/")
        aboutPanel.add(gitLabel)

        val version = String.format("<html><br><pre>gdx2d-core v%s, libgdx v%s<br>%s</pre></html>", Version.VERSION, com.badlogic.gdx.Version.VERSION, Version.COPY)
        val vLabel = JLabel(version)
      vLabel.background = Color.WHITE
        aboutPanel.add(vLabel)
    }

    /**
     * Show the about dialog as a [JOptionPane].
     */
    fun setVisible() {
        JOptionPane.showMessageDialog(parent, aboutPanel, TITLE, JOptionPane.INFORMATION_MESSAGE, icon)
    }

    companion object {

        private val TITLE = "About this application"

        private val INFO = ("<HTML><BODY>"
                + "<b>Gdx2d demos selector.</b><br><br>"
                + "Pierre-Andr&eacute; Mudry<br>"
                + "Christopher M&eacute;trailler<br><br></BODY></HTML>")

        private fun addUrl(website: JLabel, url: String, text: String) {
            website.text = String.format("<html>%s<a href=\"%s\">%s</a></html>", website.text, url, text)
            website.cursor = Cursor(Cursor.HAND_CURSOR)
            website.addMouseListener(object : MouseAdapter() {
                override fun mouseClicked(e: MouseEvent?) {
                    try {
                        Desktop.getDesktop().browse(URI(url))
                    } catch (ex: Exception) {
                        System.err.println("Cannot open $url")
                        ex.printStackTrace()
                    }

                }
            })
        }
    }
}
